pulp_2to3_migration/app/plugin/deb/pulp_2to3_models.py

```diff
@@ -0,0 +1,165 @@
+        pulp2_ids = pulp2_id_obj_map.keys()
+        pulp2_content_batch = pulp2_models.DebPackage.objects.filter(id__in=pulp2_ids).as_pymongo().only(*cls.pulp2_unit_fields)
+
+        pulp2deb_to_save = []
+        for deb in pulp2_content_batch:
+            pulp2deb_to_save.append(
+                Pulp2DebPackage(
+                    name=deb['name'],
+                    version=deb['version'],
+                    architecture=deb['architecture'],
+                    checksumtype=deb['checksumtype'],
+                    checksum=deb['checksum'],
+                    filename=deb['filename'],
+                    size=deb['size'],
+                    # Package relations:
+                    breaks=deb['control_fields'].get('Breaks', None),
+                    conflicts=deb['control_fields'].get('Conflicts', None),
+                    depends=deb['control_fields'].get('Depends', None),
+                    enhances=deb['control_fields'].get('Enhances', None),
+                    pre_depends=deb['control_fields'].get('Pre-Depends', None),
+                    provides=deb['control_fields'].get('Provides', None),
+                    recommends=deb['control_fields'].get('Recommends', None),
+                    replaces=deb['control_fields'].get('Replaces', None),
+                    suggests=deb['control_fields'].get('Suggests', None),
+                    # Control file fields:
+                    source=deb.get('source', None),
+                    section=deb.get('section', None),
+                    priority=deb.get('priority', None),
+                    installed_size=deb.get('installed_size', None),
+                    maintainer=deb.get('maintainer', None),
+                    original_maintainer=deb.get('original_maintainer', None),
+                    description=deb.get('description', None),
+                    homepage=deb.get('homepage', None),
+                    multi_arch=deb.get('multi_arch', None),
+                    # Missing?: origin, tag, bugs, essential, build_essential,
+                    # description_md5, built_using, auto_built_package
+                    pulp2content=pulp2_id_obj_map[deb['_id']],
+                )
+            )
+
+        cls.objects.bulk_create(pulp2deb_to_save, ignore_conflicts=True)
+
+    @classmethod
+    def create_pulp3_content(self):
+        """
+        Create a Pulp 3 detail Content unit for saving it later in a bulk operation.
+
+        Return an unsaved Pulp 3 Content
+        """
+        pulp3_package_fields = {
+            'package': self.name,
+            'source': self.source,
+            'version': self.version,
+            'architecture': self.architecture,
+            'section': self.section,
+            'priority': self.priority,
+            'installed_size': self.installed_size,
+            'maintainer': self.maintainer,
+            'original_maintainer': self.original_maintainer,
+            'description': self.description,
```

Comment:
@quba42 I notice that the "size" field is not present here, and I think it probably should be?

Reply:
It's plausible that creating a Package object without initializing the ```size``` field might cause problems when the uninitialized field is saved.  I would have expected a more descriptive error if that were the case, but maybe not.

Reply:
This is a good hint. I think I need to look more closely at how the `Content` and the `Artifact` classes interact. I am not sure the size field needs to be added here directly (does not appear to work that way for iso and rpm) but I may be missing some mechanism for it to be passed to the artifact that is created.

I am on holiday this week, but I will definitely pursue this when I get back, thanks.